linera-client/src/unit_tests/chain_listener.rs

```diff
@@ -142,8 +142,12 @@ async fn test_chain_listener() -> anyhow::Result<()> {
         .update_wallet_for_new_chain(chain_id0, Some(key_pair), clock.current_time())
         .await?;
     let context = Arc::new(Mutex::new(context));
-    let listener = ChainListener::new(config, context, storage);
-    listener.run().await;
+    let _handle = linera_base::task::spawn(async move {
```

Comment:
`let _ = ...` is generally a bad idea as it disables important clippy warnings.

Reply:
But that's the point here: `spawn` starts running the task immediately in the background, but it also returns a future that can be `await`ed to join the task. We don't want to join in this case, so we don't use that future, which is what Clippy would complain about.

Reply:
I see. We could also signal the listener to stop at the end of the test and wait for it cleanly.

Reply:
Yes, maybe I'll add the join handle to `ListeningClient`. Unfortunately we can't just join on `Drop`, because it's `async`. :thinking:

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3745/commits/40831db1c676f3528dc410c0bd47f250623306b6.